src/azure-cli/azure/cli/command_modules/acr/artifact_streaming.py

```diff
@@ -0,0 +1,381 @@
+    return result
+
+
+def _check_if_streaming_artifact_exists(repository,
+                                        digest,
+                                        login_server,
+                                        username,
+                                        password):
+    path = _get_referrers_path(repository, digest)
+    referrers = _obtain_referrers_from_registry(
+        login_server=login_server,
+        path=path,
+        username=username,
+        password=password,
+        artifact_type="application/vnd.azure.artifact.streaming.v1")
+    manifest = referrers['manifests']
+    return len(manifest) > 0
+
+
+def _get_last_streaming_operation(cmd,
+                                  registry_name,
+                                  repository,
+                                  digest,
+                                  tenant_suffix=None,
+                                  username=None,
+                                  password=None):
+    path = _get_v1_artifact_streaming_image_path(repository, digest)
+    # Verify if the artifact streaming create operation is already in queue
+    try:
+        login_server, username, password = get_access_credentials(
+            cmd=cmd,
+            registry_name=registry_name,
+            tenant_suffix=tenant_suffix,
+            username=username,
+            password=password,
+            repository=repository,
+            permission=RepoAccessTokenPermission.PULL.value)
+        result, _, response_status = request_data_from_registry(http_method='get',
+                                                                login_server=login_server,
+                                                                path=path,
+                                                                username=username,
+                                                                password=password)
+        if response_status == 200:
+            return result
+    except RegistryException as e:
+        if e.status_code != 404:
+            raise
+    return None
+
+
+def _is_ongoing_streaming_status(status):
+    res = status == ArtifactStreamingStatus.Running.value or status == ArtifactStreamingStatus.NotStarted.value
+    return res
+
+
+def _create_streaming_artifact_poller(repository,
+                                      operationId,
+                                      create_response,
+                                      login_server,
+                                      username,
+                                      password):
+    path = _get_v1_artifact_streaming_operation_path(repository, operationId)
+    timmer = 5
+    time_count = 0
+    progress = "0%"
+    status = ArtifactStreamingStatus.NotStarted.value
+    while _is_ongoing_streaming_status(status):
+        time.sleep(timmer)
+        time_count += timmer
+        if timmer < 21:
+            timmer += 1
+        poller, _, _ = request_data_from_registry(http_method='get',
```

Comment:
Does every request will do the auth request to get the access token first and then send the request?

Reply:
No, we get one token prior to the poller and use that to do the call. It might cause problems for operations running over 3 Hrs since according to Cesar that's the life expectancy of the token I'm generating for this request.